cgsmiles/tests/test_cgsmile_parsing.py

```diff
@@ -3,68 +3,96 @@
+                        [(0, 1), (1, 2)],
+                        [1, 1]),
+                        # smiple linear sequenece with multi-edge
+                        ("{[#PMA]1[#PEO]1}",
+                        ["PMA", "PEO"],
+                        [(0, 1)],
+                        [2]),
                         # simple branched sequence
                         ("{[#PMA][#PMA]([#PEO][#PEO])[#PMA]}",
                         ["PMA", "PMA", "PEO", "PEO", "PMA"],
-                        [(0, 1), (1, 2), (2, 3), (1, 4)]),
+                        [(0, 1), (1, 2), (2, 3), (1, 4)],
+                        [1, 1, 1, 1]),
                         # simple sequence two branches
                         ("{[#PMA][#PMA][#PMA]([#PEO][#PEO])([#CH3])[#PMA]}",
                         ["PMA", "PMA", "PMA", "PEO", "PEO", "CH3", "PMA"],
-                        [(0, 1), (1, 2), (2, 3), (3, 4), (2, 5), (2, 6)]),
+                        [(0, 1), (1, 2), (2, 3), (3, 4), (2, 5), (2, 6)],
+                        [1, 1, 1, 1, 1, 1]),
                         # simple linear sequence with expansion
                         ("{[#PMA]|3}",
                         ["PMA", "PMA", "PMA"],
-                        [(0, 1), (1, 2)]),
-                        # smiple cycle seqeunce
+                        [(0, 1), (1, 2)],
+                        [1, 1]),
+                        # smiple cycle sequence
                         ("{[#PMA]1[#PEO][#PMA]1}",
                         ["PMA", "PEO", "PMA"],
-                        [(0, 1), (1, 2), (0, 2)]),
+                        [(0, 1), (1, 2), (0, 2)],
+                        [1, 1, 1]),
+                        # smiple cycle sequence with %
+                        ("{[#PMA]%123[#PEO][#PMA]%123}",
+                        ["PMA", "PEO", "PMA"],
+                        [(0, 1), (1, 2), (0, 2)],
+                        [1, 1, 1]),
                         # complex cycle
                         ("{[#PMA]1[#PEO]2[#PMA]1[#PEO]2}",
                         ["PMA", "PEO", "PMA", "PEO"],
-                        [(0, 1), (1, 2), (0, 2), (1, 3), (2, 3)]),
-                        # complex cycle
+                        [(0, 1), (1, 2), (0, 2), (1, 3), (2, 3)],
+                        [1, 1, 1, 1, 1]),
+                        # complex cycle with %
+                        ("{[#PMA]%134[#PEO]%256[#PMA]%134[#PEO]%256}",
+                        ["PMA", "PEO", "PMA", "PEO"],
+                        [(0, 1), (1, 2), (0, 2), (1, 3), (2, 3)],
+                        [1, 1, 1, 1, 1]),
+                        # complex cycle with three times same ID
                         ("{[#PMA]1[#PEO]2[#PMA]1[#PEO]2[#PMA][#PMA]1}",
                         ["PMA", "PEO", "PMA", "PEO", "PMA", "PMA"],
                         [(0, 1), (1, 2), (0, 2), (1, 3), (2, 3), (3, 4),
-                         (4, 5), (0, 5)]),
+                         (4, 5), (0, 5)],
+                        [1, 1, 1, 1, 1, 1, 1, 1]),
```

Comment:
This should be an error, since you have a leftover `1`
If you want multiple cycles:
`{[#PMA]13[#PEO]2[#PMA]1[#PEO]2[#PMA][#PMA]3}`